.expeditor/scripts/end_to_end/run_e2e_test_core.ps1

```diff
@@ -0,0 +1,126 @@
+        (Get-Item -Path $Path).LastWriteTime -ge $Time
+    }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for '$Path' to be updated after '$($Time.ToString("yyyy-MM-ddTHH:MM:ssZ"))'" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+}
+
+function Test-Port([string]$ComputerName, [int]$Port) {
+    if (Get-Command Test-NetConnection -ErrorAction SilentlyContinue) {
+        (Test-NetConnection -ComputerName $ComputerName -Port $Port).TcpTestSucceeded
+    }
+    else {
+        if ($IsWindows) {
+            Write-Error "`Test-NetConnection` command was not found and `Test-Connection` does not work correctly on Windows"
+        }
+        Test-Connection -ComputerName $ComputerName -TCPPort $Port
+    }
+}
+
+function Wait-Supervisor($Timeout = 1) {
+    Write-Host "Waiting up to $Timeout seconds for Supervisor to start..."
+    $testScript = { Test-Port -ComputerName localhost -Port 9631 }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for Supervisor to start" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+    Write-Host "Supervisor is now running."
+}
+
+function Start-Supervisor($Timeout = 1) {
+    hab sup run --no-color &
+    Wait-Supervisor -Timeout $Timeout
+}
+
+function Wait-SupervisorService($ServiceName, $Timeout = 1) {
+    Write-Host "Waiting up to $Timeout seconds for Supervisor to start $ServiceName ..."
+    $testScript = { 
+        try {
+            $status = (Invoke-WebRequest "http://localhost:9631/services/$ServiceName/default" |
+                ConvertFrom-Json).process.state
+            $status -eq "up"
+        }
+        catch { } # We ignore 404s and other unsuccesful codes
+    }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for Supervisor to start $ServiceName" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+    Write-Host "$ServiceName is now up."
+}
+  
+function Install-SupervisorService($PackageName, $Timeout = 1) {
+    $svcName = ($PackageName -split "/")[1]
+    $_ = hab svc load $PackageName
+    Wait-SupervisorService $svcName -Timeout $Timeout
+    $svcName
+}
+  
+function Wait-SupervisorServiceUnload($ServiceName, $Timeout = 1) {
+    Write-Host "Waiting up to $Timeout seconds for Supervisor to unload $ServiceName ..."
+    $testScript = { 
+        try {
+            Invoke-WebRequest "http://localhost:9631/services/$ServiceName/default"
+        }
+        catch {
+            $statusCode = $_.Exception.Response.StatusCode.value__
+            $statusCode -eq 404
+        }
+    }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for Supervisor to unload $ServiceName" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+    Write-Host "$ServiceName is now unloaded."
+}
+
+function Uninstall-SupervisorService($PackageName, $Timeout = 1) {
```

Comment:
`unload` instead of `uninstall` since we are not actually uninstalling.

Reply:
Yeah, I was torn between using powershells accepted verbs and useing the habitat vocabulary. I will go ahead and switch it to match habitat.

Reply:
Glad to see you are trying to use the PS verbs! If we wanted to be anal about that here we could actually uninstall too but up to you.